### ISSUE TITLE
Deduplication doesn't occur with `imports_granularity = "Item"

### ISSUE THREAD

**Issue by djmarcin** [rust-lang/rustfmt#4725]
**Describe the bug**

Duplicate imports are not removed with `imports_granularity = "Item"` turned on.

**To Reproduce**

test.rs
```
use std::env;
use std::env;

fn main() {
  println!("{:?}", env::var("TEST"));
}
```

.rustfmt.toml
```
unstable_features = true
edition = "2018"
imports_granularity = "Item"
```

Run `rustfmt test.rs`

**Expected behavior**

Duplicate `use std::env;` should be removed. It is not.

**Meta**

- rustfmt version: `rustfmt 1.4.36-nightly (7de6968 2021-02-07)`
- From where did you install rustfmt?:  `asdf install` with .tool-versions `rust nightly-2021-02-17`
- How do you run rustfmt: `rustfmt`, in VSCode

**Comment by calebcartwright** [comment]
Thanks for opening this! We'd actually just observed this recently as well (https://github.com/rust-lang/rustfmt/pull/4716#discussion_r581608452) so good to have an issue to track :+1:

**Comment by calebcartwright** [comment]
For anyone interested in working on this, would suggest starting here given the current code structure. Note that the trees are _not_ yet sorted at this point, as sorting conditionally happens a bit later on depending on the user specified value for the `reorder_imports` option. We need to dedupe regardless of the value of `reorder_imports`

https://github.com/rust-lang/rustfmt/blob/74768b7f599278f72bc954b22d9ec214976eee50/src/formatting/imports.rs#L185-L201

**Comment by zaki-yama** [comment]
@calebcartwright 
Hello! May I try this?

**Comment by calebcartwright** [comment]
> @calebcartwright
> Hello! May I try this?

Sure! However, I think @davidBar-On may have beaten you to it in #4737. Will leave it to the two of you to sort who wants to work on it.

**Comment by davidBar-On** [comment]
@zaki-yama, I suggest to see if my approach in PR #4737 to resolve the issue is accepted.  If not, I have no problem that you will take over. (Usually I am not trying to resolve new `help wanted` assigned issues, but this one is somewhat related to #4681 which I submitted a PR for.)

**Comment by zaki-yama** [comment]
@calebcartwright @davidBar-On 
Thank you for replying.
Sure! Since I'm new to try the issue in this repository, I'll see and learn from #4737.
Also, I'll see other `help wanted` or `good first issue` labeled issues.

**Comment by calebcartwright** [comment]
Resolved via 6b415bd23a98531d26a213288cc60ffd0e898ef3, backport/release pending
